commit eb6a4dcce33925ac95023bbe5199474f8db40ba7
Author: Will Deacon <will@kernel.org>
Date:   Fri Aug 23 13:03:55 2019 +0100

    arm64: mm: Add ISB instruction to set_pgd()
    
    Commit 6a4cbd63c25a ("Revert "arm64: Remove unnecessary ISBs from
    set_{pte,pmd,pud}"") reintroduced ISB instructions to some of our
    page table setter functions in light of a recent clarification to the
    Armv8 architecture. Although 'set_pgd()' isn't currently used to update
    a live page table, add the ISB instruction there too for consistency
    with the other macros and to provide some future-proofing if we use it
    on live tables in the future.
    
    Reported-by: Mark Rutland <mark.rutland@arm.com>
    Reviewed-by: Mark Rutland <mark.rutland@arm.com>
    Signed-off-by: Will Deacon <will@kernel.org>

diff --git a/arch/arm64/include/asm/pgtable.h b/arch/arm64/include/asm/pgtable.h
index feda729..2faa776 100644
--- a/arch/arm64/include/asm/pgtable.h
+++ b/arch/arm64/include/asm/pgtable.h
@@ -605,6 +605,7 @@ static inline void set_pgd(pgd_t *pgdp, pgd_t pgd)
 
 	WRITE_ONCE(*pgdp, pgd);
 	dsb(ishst);
+	isb();
 }
 
 static inline void pgd_clear(pgd_t *pgdp)
